libfreerdp/core/gateway/http.c

```diff
@@ -302,12 +340,50 @@ BOOL http_context_set_rdg_auth_scheme(HttpContext* context, const char* RdgAuthS
 	return context->RdgAuthScheme != NULL;
 }
 
+BOOL http_context_set_cookie(HttpContext* context, const char* CookieName, const char* CookieValue)
+{
+	BOOL status;
+	if (!context || !CookieName || !CookieValue)
+		return FALSE;
+	if (ListDictionary_Contains(context->cookies, CookieName))
+	{
+		char* dupCookieValue = strdup(CookieValue);
```

Comment:
you can eliminate all these `strdup` things by just providing a `fnObjectNew` for the list key/value objects